src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -59,6 +59,7 @@
             "Indentation",
             "JavadocMethod",
             "MissingJavadocType",
+            "NoCodeInFile",
```

Comment:
Look at top text, it says are "not compatible ... till ....". This check will never be compatible. It always logs by line number since there is never any AST.

We need to follow pattern of other checks where we couldn't support them, `Regexp` and `RegexpSinglelineJava` and give a reason. xdoc should also be updated too.

Reply:
Done.